cookbook-react/src/hooks/useCollapse/index.tsx

```diff
@@ -0,0 +1,81 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+
+const DEFAULT_COLLAPSE_TIME = 300;
+
+export interface CollapsibleParameters {
+  defaultIsOpen?: boolean;
+  onChange?: (isOpen: boolean) => void;
+  animateOnCollapseEffect?: boolean;
+  collapseTime?: number;
+}
+
+const useCollapse = <Element extends HTMLElement>({
+  defaultIsOpen = true,
+  onChange,
+  animateOnCollapseEffect = true,
+  collapseTime = DEFAULT_COLLAPSE_TIME
+}: CollapsibleParameters) => {
+  const [collapsed, setCollapsed] = useState(!defaultIsOpen);
+  const [originalHeight, setOriginalHeight] = useState(0);
+  const collapsibleRef = useRef<Element>(null);
+
+  const setCollapsedStyles = (height: number, isCollapsed: boolean) => {
+    const { style } = collapsibleRef.current!;
+    const heightString = isCollapsed ? '0' : `${height}px`;
+
+    style.height = heightString;
+
+    style.paddingTop = isCollapsed ? '0' : '';
+    style.paddingBottom = isCollapsed ? '0' : '';
+    style.marginTop = isCollapsed ? '0' : '';
+    style.marginBottom = isCollapsed ? '0' : '';
+  };
+
+  const calculateHeight = useCallback(() => {
+    const { scrollHeight } = collapsibleRef.current!;
+
+    const newHeight = scrollHeight ?? 0;
+    setOriginalHeight(newHeight);
+  }, []);
+
+  // Sets initial height values
+  useEffect(() => {
+    collapsibleRef.current!.style.transition = `height ${collapseTime}ms ease-out, padding ${collapseTime}ms ease-out, margin ${collapseTime}ms ease-out`;
+    collapsibleRef.current!.style.overflow = 'hidden';
+
+    calculateHeight();
```

Comment:
```suggestion
    collapsibleRef.current!.style.transition = `height ${collapseTime}ms ease-out, padding ${collapseTime}ms ease-out, margin ${collapseTime}ms ease-out`;
    collapsibleRef.current!.style.overflow = 'hidden';

    calculateHeight();
```
What if current is not defined yet 🤔 , why don't you add an if?